File: Projet/app/src/main/java/utbm/lepysidawy/code_p25/dao/RaceDAO.java
package utbm.lepysidawy.code_p25.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import utbm.lepysidawy.code_p25.entity.Race;

/**
 * DAO used to interact with the Race table
 */
@Dao
public interface RaceDAO {

    @Query("SELECT * FROM Race")
    List<Race> getAll();

    @Query("SELECT RACE.NAME FROM Race WHERE RACE.idRace = :raceId")
    String getRaceName(int raceId);

    @Insert
    long insert(Race race);

    @Delete
    void delete(Race race);

}

File: Projet/app/app/src/main/java/com/codep25/codep25/model/storage/dao/RaceDao.kt
package com.codep25.codep25.model.storage.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.codep25.codep25.model.entity.*

@Dao
interface RaceDao {

    @Query("SELECT * FROM race")
    fun getRaces(): List<Race>

    @Query("SELECT * FROM race WHERE race_id == :id")
    fun getRaceById(id: Long): Race

    @Query("SELECT * FROM racing_team WHERE race_id == :id")
    fun getRacingTeamsByRaceId(id: Long): List<RacingTeamEntity>

    @Query("SELECT * FROM racing_team_link WHERE team_id == :id")
    fun getParticipantsForTeam(id: Long): List<RacingTeamLinkEntity>

    @Query("SELECT * FROM race_time")
    fun getTimes(): List<RaceTimeEntity>

    @Query("SELECT * FROM race_time WHERE team_id == :id")
    fun getTimesForTeam(id: Long): List<RaceTimeEntity>

    @Insert
    fun insertRace(race: Race): Long

    @Insert
    fun insertRacingTeam(racingTeamEntity: RacingTeamEntity): Long

    @Insert
    fun insertRacingTeamLinks(vararg racingTeamLinkEntity: RacingTeamLinkEntity)

    @Insert
    fun insertRaceTimes(vararg raceTimeEntity: RaceTimeEntity)
}